codec-http2/src/test/java/io/netty/handler/codec/http2/Http2MultiplexTest.java

```diff
@@ -1367,6 +1367,33 @@ public void windowUpdatesAreFlushed() {
         assertTrue(flushSniffer.checkFlush());
     }
 
+    @Test
+    public void closeStreamWithoutFlush() {
+        Http2StreamChannel childChannel3 = newInboundStream(3, true, new LastInboundHandler());
+        Http2StreamChannel childChannel5 = newInboundStream(5, true, new LastInboundHandler());
+        FlushSniffer flushSniffer = new FlushSniffer();
+        parentChannel.pipeline().addFirst(flushSniffer);
+
+        //1. write data and batch flush
+        byte[] dataBuf = "test".getBytes();
+        childChannel5.write(new DefaultHttp2DataFrame(Unpooled.wrappedBuffer(dataBuf), false));
+        childChannel3.write(new DefaultHttp2DataFrame(Unpooled.wrappedBuffer(dataBuf), false));
+        childChannel3.flush();
+        assertTrue(flushSniffer.checkFlush());
+
+        //2. write end header frame
+        Http2Headers headers = new DefaultHttp2Headers();
+        headers.set("test", "test");
+        childChannel3.write(new DefaultHttp2HeadersFrame(headers, true));
+        childChannel5.write(new DefaultHttp2HeadersFrame(headers, true));
+
+        //do not auto flush, it will be 'false'
+        assertFalse(flushSniffer.checkFlush());
+        //batch flush
+        childChannel5.flush();
```

Comment:
Btw, have you considered adding `FlushConsolidationHandler` between `Http2ConnectionHandler` and `Http2MultiplexHandler`?

Reply:
> Btw, have you considered adding `FlushConsolidationHandler` between `Http2ConnectionHandler` and `Http2MultiplexHandler`?

I tried, this way can reduce `flush`. thks

Reply:
> @icodening taking the above description into account, can you clarify the exact flow of how you plan to "batch flush" on a closed stream?

I will write the message to the `queue` and schedule the eventloop to execute. 
 `ctx.write` will be called when the number of messages is less than the threshold,  and `ctx.flush` will be called when the number of messages is more than the threshold.

Reply:
> @normanmaurer it sounds like it's incorrect to invoke childChannel5.flush() here, because childChannel5.isActive() returns false at this point. However, there is no other easy way for users of StreamChannel API to flush after response is written except using writeAndFlush, which complicates usability. In ST, we also invoke flush() after the last write(...) because we have a separate logic to control flushes.

It seems more reasonable to use the `parent channel` flush